### PR TITLE
 Correct Inconsistencies of Timeseries for Transactions

### PR DESCRIPTION
**Pull request by SudharakaP**
Related to https://github.com/opencollective/opencollective/issues/4995

Related Front-end PR: https://github.com/opencollective/opencollective-frontend/pull/7193

### CODE REVIEW

File: server/graphql/v2/object/Host.js [hunk 1]
```diff
@@ -646,11 +657,13 @@ export const Host = new GraphQLObjectType({
           const expenseAmountOverTime = async () => {
             const dateFrom = args.dateFrom ? moment(args.dateFrom).toISOString() : undefined;
             const dateTo = args.dateTo ? moment(args.dateTo).toISOString() : undefined;
+            const timeUnit = getTimeUnit(numberOfDays);
```
Comment (Betree) [inline server/graphql/v2/object/Host.js:660]: What happens to `args.timeUnit`? Sounds to me like we should remove its `defaultValue`, do something like:

```suggestion
            const timeUnit = args.timeUnit || getTimeUnit(numberOfDays);
```

And clarify all that in the field description.

Comment (SudharakaP) [reply server/graphql/v2/object/Host.js]: I've changed this to make sure the default value is removed and made the changes. Thanks. 🙂

File: server/graphql/v2/object/Host.js [hunk 2]
```diff
@@ -67,6 +67,16 @@ const getNumberOfDays = (startDate, endDate, host) => {
...
 };
 
+const getTimeUnit = numberOfDays => {
+  if (numberOfDays <= 7) {
+    return 'WEEK';
+  } else if (numberOfDays <= 366) {
+    return 'MONTH';
+  } else {
+    return 'YEAR';
+  }
```
Comment (Betree) [inline server/graphql/v2/object/Host.js:77]:
```suggestion
  if (numberOfDays <= 21) {
    return 'DAY'; // Up to 3 weeks
  } else if (numberOfDays <= 90) {
    return 'WEEK'; // Up to 3 months
  } else if (numberOfDays <= 365 * 3) {
    return 'MONTH'; // Up to 3 years
  } else {
    return 'YEAR';
  }
```

**Review by Betree — COMMENTED** [review]
We can always improve them later, but the following values for the time unit would make more sense. They'll prevent having a single column in the chart, which is something we should avoid.

Basically, we want the chart to display meaningful data regardless of the selected time range.

File: server/graphql/v2/object/Host.js (hunk 2 above)
Comment (SudharakaP) [reply server/graphql/v2/object/Host.js]: @Betree : The requirement to show the weekly amounts seems a bit complicated to wrap my head around. For now I've added,

https://github.com/opencollective/opencollective-api/blob/34c1e2a719caa1f7f56241af27cd674aeaee2390/server/graphql/v2/object/Host.js#L71-L77

I think that's better that what was there hopefully. If no objections, I'll do this weekly amount display in a separate PR? 🤔

Comment (Betree) [reply server/graphql/v2/object/Host.js]: If you do `numberOfDays <= 31` you'll end up with the same issue of having a chart that only contains a single value. Play with it to find something that looks nice, but something like `90` sounds like a decent default.

No problem to add `WEEK` in a separate PR if it complicates things (and actually, we should only add it if there's a value in doing so). Could you shortly explain where the complexity to add this is coming from?

Comment (SudharakaP) [reply server/graphql/v2/object/Host.js]: Actually looking at this today I realized the front-end implementation for this dynamic timeunits can be greatly simplified. I've just did that and now it should look much better and also I've added the week support. 🙂 Sorry about this, I feel I was making a too complicated mess out of this. 🤔

**Comment by znarf** [comment]
This is simply crashing. 

Screenshot 2021-12-06 at 10 58 31

**Review by znarf — CHANGES_REQUESTED** [review]
The page should not be crashing (might be frontend side).

**Comment by SudharakaP** [comment]
> The page should not be crashing (might be frontend side).

Yes this is my mistake on the front end side. I've corrected it.

**Review by Betree — APPROVED** [review]
Works for me, we'll see with experience if the default values for `timeUnit` make sense

**Pull request merged**